config/structure.go

```diff
@@ -0,0 +1,85 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type BotConfig struct {
+	InteractiveProgress bool        `json:"interactiveProgress"`
+	Servers             []VCSServer `json:"servers"`
+}
+
+type staticVCSServerFields struct {
+	Type           string         `json:"type"`
+	BaseURL        string         `json:"baseUrl"`
+	SyncTag        string         `json:"syncTag"`
+	SSHCreds       SSHCredentials `json:"sshCreds"`
+	ProjectsToSync []VCSProject   `json:"projectsToSync"`
+}
+
+type VCSServer struct {
+	staticVCSServerFields
+	APIToken CredentialSource
+}
+
+func (svr *VCSServer) UnmarshalJSON(bytes []byte) error {
+	type IntermediateVCSServer struct {
+		staticVCSServerFields
+		APIToken json.RawMessage `json:"apiToken"`
+	}
+	var intermediateRepresentation IntermediateVCSServer
+	if parseErr := json.Unmarshal(bytes, &intermediateRepresentation); parseErr != nil {
+		return fmt.Errorf("failed to read vcs server: %w", parseErr)
+	}
+
+	svr.staticVCSServerFields = intermediateRepresentation.staticVCSServerFields
+
+	if tokenParseErr := unmarshalCredentialSource(intermediateRepresentation.APIToken, &svr.APIToken); tokenParseErr != nil {
+		return fmt.Errorf("failed to read api token for server %v: %w", intermediateRepresentation.BaseURL, tokenParseErr)
+	}
+	return nil
+}
+
+type SSHCredentials struct {
+	IsPresent  bool
+	PublicKey  CredentialSource
+	PrivateKey CredentialSource
+}
+
+func (pc *SSHCredentials) UnmarshalJSON(bytes []byte) error {
```

Comment:
Change the receiver name here since it doesn't match up with the type name (the type got renamed)